### ISSUE TITLE
Add option to set SDK-global attributes, common to all features

### ISSUE THREAD

**Issue by JetForMe** [DataDog/dd-sdk-ios#721]
I have to create and pass an `AppContext` when initializing Datadog, but I can't initialize it with my own values (which I want to do to give more context to the app, since there's no obvious way to set global logging attributes).

Unfortunately, the memberwise initalizer is `internal`. I'd like to do this:

```swift
	let appCTX = Datadog.AppContext(bundleType: .iOSApp,
					bundleIdentifier: Bundle.main.bundleIdentifier,
					bundleVersion: DevUtils.versionString,
					bundleName: Bundle.main.object(forInfoDictionaryKey: "CFBundleExecutable") as? String)
        Datadog.initialize(appContext: appCTX,
                           trackingConsent: .granted,
                           configuration: datadogConfig)
```

**Comment by ncreated** [comment]
Hello @JetForMe 👋.

> I can't initialize it with my own values (which I want to do to give more context to the app, since there's no obvious way to set global logging attributes)

Custom attributes can be set in all our products. When it comes to logging you can do:
```swift
logger.addAttribute(forKey: "dev-version", value: DevUtils.versionString)
```
and the `dev-version` attribute will be added to all logs you send through this instance of `logger`. You can further [create facet](https://docs.datadoghq.com/logs/explorer/facets/) on `dev-version` and use it as an alternative to our reserved `version` attribute. Isn't this what you're looking for?

You can find more on advanced logs collection and attributes in [this doc](https://docs.datadoghq.com/logs/log_collection/ios/?tab=cocoapods#global-attributes).

> Unfortunately, the memberwise initalizer is internal.

`AppContext` has [public](https://github.com/DataDog/dd-sdk-ios/blob/bda05a0017e2bc832ccab4ca0565ba335d5be8e8/Sources/Datadog/Datadog.swift#L22) initializer which you can use to overwrite the `Bundle`. This might be useful if instantiating the SDK in other bundles than `.main` (e.g. in `Bundle.module`). And we don't have an API for direct manipulation of these values, as they are used internally for various things (including reserved `version` attribute and SDK controlling performance in app extensions).

**Comment by JetForMe** [comment]
I need to set _global_ values. I have many loggers (basically one per class) throughout my code, and I want to set some values for all of them in one place. In this case, your version that you set is insufficient, and I'd like to set that myself, but I can set an additional one (I need to include things like the build number and whether or not it's a debug or release build).

**Comment by ncreated** [comment]
Hey @JetForMe 👋. Sorry for late response. This is a feature request and we may consider this in the future if more people ask for it. Meanwhile, please note that setting "global SDK" attributes, common to all features is possible with following:

```swift
let commonAttributes: [AttributeKey: AttributeValue] = [
    "foo": 1,
    "bar": "bizz"
]

commonAttributes.forEach { key, value in
    logger1.addAttribute(forKey: key, value: value)
    logger2.addAttribute(forKey: key, value: value)
    Global.rum.addAttribute(forKey: key, value: value)
}
```
I understand though that this might be not super handy. Sharing only as a workaround 🙂.

**Comment by ncreated** [comment]
Solved in [**2.8.1**](https://github.com/DataDog/dd-sdk-ios/releases/tag/2.8.1) with adding support to global log attributes. See: https://github.com/DataDog/dd-sdk-ios/pull/1707
